src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+import axios from "axios";
+import { GitHubServerApp } from "models/github-server-app";
+
+export const JiraServerUrlPost = async (
+	req: Request,
+	res: Response
+): Promise<void> => {
+	// todo - validate url
+	const { gheServerURL, installationId } = req.body;
+	req.log.info(`Verifying provided GHE server url: ${gheServerURL}`, installationId);
+
+	try {
+		const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
+
+		if (gitHubServerApps?.length) {
+			req.log.info(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
+			res.status(200).send({ moduleKey: "github-list-apps-page" });
+		} else {
+			req.log.info(`No existing GitHub apps found for url: ${gheServerURL}. Redirecting to Jira app creation page.`);
+
+			const isValid = await axios.get(gheServerURL);
+			req.log.info(`Successfully verified GHE server url: ${gheServerURL}`, isValid);
+
+			res.status(200).send({ success: true, moduleKey: "github-app-creation-page"  });
+		}
+	} catch (err) {
+		req.log.error(`Something went wrong: ${gheServerURL}`);
+		// TODO - adding error mapping (create confluence page with error codes)
```

Comment:
Going to add the partial as part of [ARC 1473](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a&selectedIssue=ARC-1473)

Reply:
Will tackle mapping errors as part of [ARC 1473](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a&selectedIssue=ARC-1473)